src/buildstream_plugins/sources/cargo.py

```diff
@@ -375,7 +375,7 @@ def get_ref(self):
 
     def set_ref(self, ref, node):
         node["ref"] = ref
-        self._recompute_crates(ref)
+        self._recompute_crates(node.get_sequence("ref"))
```

Comment:
Why is this ncessary? Does ref change to somthing else after assign?

Reply:
Yes.

After assigning a python `list` object as a member of a `MappingNode`, it will become a `SequenceNode` instead.

Reply:
Surprising semantics for dict-like object but okay.